jackbox-utility: init at 1.3.7+1

jpeterburs opened the pull request:
## Description of changes

JackboxUtility is an app to download patches and launch Jackbox games.
With this app, one is able to download translation patches for the Jackbox games.

GitHub: https://github.com/jackboxutility/jackboxutility
Homepage: https://jackboxutility.com/

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jpeterburs committed:
maintainers: add jpeterburs

jpeterburs committed:
jackbox-utility: init at 1.3.7+1

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+  '';
+
+  meta = with lib; {
+    description = "An app to download patches and launch Jackbox games ";
```
```suggestion
    description = "An app to download patches and launch Jackbox games";
```

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  meta = with lib; {
+    description = "An app to download patches and launch Jackbox games ";
+    homepage = "https://github.com/JackboxUtility/JackboxUtility";
```
```suggestion
    homepage = "https://jackboxutility.com";
```

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+  meta = with lib; {
+    description = "An app to download patches and launch Jackbox games ";
+    homepage = "https://github.com/JackboxUtility/JackboxUtility";
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Plus;
```

`gpl3` is deprecated.

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+    homepage = "https://github.com/JackboxUtility/JackboxUtility";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ jpeterburs ];
+    mainProgram = "jackbox-utility";
```
```suggestion
    mainProgram = "jackbox-utility";
    platforms = [ "x86_64-linux" ];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+  '';
+
+  postInstall = ''
+    cat << EOF > $out/share/applications/${pname}.desktop
```
You can do this:

```nix
  desktopItems = [
    (makeDesktopItem {
      name = "JackboxUtility";
      exec = "jackbox-utility";
      comment = "An app to download patches and launch Jackbox games";
      desktopName = "JackboxUtility";
      categories = [ "Game" ];
    })
  ];
```

And add `copyDesktopItems` to `nativeBuildInuts`

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+      Type=Application
+      Name=JackboxUtility
+      Exec=$out/bin/${pname}
+      Icon=$out/bin/data/flutter_assets/assets/logo.png
```
It's better to link icon to a standard path, e.g. `$out/share/pixmaps/jackbox-utility.png` and write `icon = jackbox-utility`

Aleksanaa commented on pkgs/by-name/ja/jackbox-utility/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+    mkdir -p $out/bin $out/share/applications
+    unzip -d $out/bin $src
+    mv $out/bin/JackboxUtility $out/bin/${pname}
```
Please do not reference `pname` unnecessarily.

jpeterburs replied:
Thought it would be better to reference than to repeat it, but makes sense

Aleksanaa reviewed (approved):
Looks good, but would be better to build from source

jpeterburs committed:
jackbox-utility: refactor

Apply suggestions from code-review by @Aleksanaa

jpeterburs commented:
I'm new to this, and I wanted to use this software in my NixOS installation in the future, and don't really know that much about Dart and/or flutter.

jpeterburs commented:
Version `1.3.7+2` has been release.

Full Changelog: https://github.com/JackboxUtility/JackboxUtility/releases/tag/1.3.7%2B2

jpeterburs committed:
jackbox-utility: 1.3.7+1 -> 1.3.7+2

Diff: https://github.com/JackboxUtility/JackboxUtility/compare/1.3.7+1...1.3.7+2
Changelog: https://github.com/JackboxUtility/JackboxUtility/releases/tag/1.3.7%2B2
